### PR TITLE
Added tests, psalm and phpcs fixes

### LINKED ISSUE #10: Code quality

### LINKED ISSUE #11: codecov badge

### LINKED ISSUE #13: Bump `psr/http-message` to: `^1.0 || ^2.0`

### LINKED ISSUE #14: green badge
https://insight.symfony.com/projects/94ace687-5124-446f-a324-0ecca1b47f88

[SymfonyInsight](https://insight.symfony.com/projects/94ace687-5124-446f-a324-0ecca1b47f88)

### PR DESCRIPTION
**Pull request by Jurj-Bogdan**
#10

### CODE REVIEW

**Comment by Jurj-Bogdan** [comment]
@arhimede a symfony insight badge would be needed

File: README.md [hunk 1]
```diff
@@ -2,12 +2,15 @@
...
 
 ![OSS Lifecycle](https://img.shields.io/osslifecycle/dotkernel/dot-flashmessenger)
-![PHP from Packagist (specify version)](https://img.shields.io/packagist/php-v/dotkernel/dot-flashmessenger/3.2.0)
+![PHP from Packagist (specify version)](https://img.shields.io/packagist/php-v/dotkernel/dot-flashmessenger/3.4.0)
 
 [![GitHub issues](https://img.shields.io/github/issues/dotkernel/dot-flashmessenger)](https://github.com/dotkernel/dot-flashmessenger/issues)
 [![GitHub forks](https://img.shields.io/github/forks/dotkernel/dot-flashmessenger)](https://github.com/dotkernel/dot-flashmessenger/network)
 [![GitHub stars](https://img.shields.io/github/stars/dotkernel/dot-flashmessenger)](https://github.com/dotkernel/dot-flashmessenger/stargazers)
-[![GitHub license](https://img.shields.io/github/license/dotkernel/dot-flashmessenger)](https://github.com/dotkernel/dot-flashmessenger/blob/3.2.0/LICENSE.md)
+[![GitHub license](https://img.shields.io/github/license/dotkernel/dot-flashmessenger)](https://github.com/dotkernel/dot-flashmessenger/blob/3.4.0/LICENSE.md)
```
Comment (alexmerlin) [inline README.md]:
```suggestion
[![GitHub license](https://img.shields.io/github/license/dotkernel/dot-flashmessenger)](https://github.com/dotkernel/dot-flashmessenger/blob/3.0/LICENSE.md)
```

File: composer.json [hunk 2]
```diff
@@ -15,16 +15,24 @@
...
+        "laminas/laminas-servicemanager": "^3.21.0",
+        "laminas/laminas-session": "^2.16.0",
+        "mezzio/mezzio-template": "^2.8.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "^9.5.20",
-        "squizlabs/php_codesniffer": "^3.6.2",
-        "mezzio/mezzio-template": "^2.4.0"
+        "laminas/laminas-coding-standard": "^2.5.0",
+        "mikey179/vfsstream": "^v1.6.11",
```
Comment (alexmerlin) [inline composer.json]:
```suggestion
```

File: test/FlashMessengerTest.php [hunk 3]
```diff
@@ -1,20 +1,119 @@
...
     {
-        //...
+        $this->subject->addInfo('info message', 'success-channel');
+        $this->subject->addSuccess('success message', 'success-channel');
+        $this->subject->addWarning(['warning message'], 'error-channel');
+        $this->subject->addError(['first error message', 'second error message'], 'error-channel');
+
+        $ref       = new ReflectionObject($this->subject);
+        $container = $ref->getProperty('sessionContainer');
+        $container->setAccessible(true);
```
Comment (alexmerlin) [inline test/FlashMessengerTest.php]:
```suggestion
```

File: test/FlashMessengerTest.php [hunk 4]
```diff
@@ -1,20 +1,119 @@
...
+        );
+    }
+
+    public function testDataAddingAndRetrieval(): void
+    {
+        $this->subject->addData('test key', 'test data');
+
+        $ref       = new ReflectionObject($this->subject);
+        $container = $ref->getProperty('sessionContainer');
+        $container->setAccessible(true);
```
Comment (alexmerlin) [inline test/FlashMessengerTest.php]:
```suggestion
```

**Review by alexmerlin — CHANGES_REQUESTED** [review]
Please restore file permissions (`755` -> `644`) for the following files:
* `.gitignore`
* `CHANGELOG.md`
* `LICENSE.md`
* `OSSMETADATA`
* `README.md`
* `composer.json`

**Review by alexmerlin — CHANGES_REQUESTED** [review]
I just noticed that the permissions have changed on the files in the `src` directory too.
Please restore those as well (`755` -> `644`).

**Review by alexmerlin — APPROVED** [review]

**Pull request merged**